File: algorithms/selectionSort/practice/SelectionSort1.java
import java.util.Arrays;

public class SelectionSort1 {
    public static void main(String[] args) {
        int[] arr = new int[] { 2, 1, 4, 3, 7, 3 };
        selectionSort(arr);
        Arrays.stream(arr).forEach(System.out::println);
    }

    private static void selectionSort(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            int j = i + 1;
            int smallestIndex = i;
            while (j < arr.length) {
                if (arr[j] < arr[smallestIndex])
                    smallestIndex = j;
                j++;
            }

            if (i != smallestIndex) {
                arr[i] ^= arr[smallestIndex];
                arr[smallestIndex] ^= arr[i];
                arr[i] ^= arr[smallestIndex];
            }
        }
    }
}

File: python/algorithms/sort/selection_sort/practice/selection_sort_1.py
def selection_sort(arr):
    for i in range(0, len(arr)):
        smallest = i
        for j in range(i + 1, len(nums)):
            if arr[j] < arr[smallest]:
                smallest = j
                
        if i is not smallest:
            arr[i] ^= arr[smallest]
            arr[smallest] ^= arr[i]
            arr[i] ^= arr[smallest]
    return arr

nums = [2, 5, 8, 3, 9, 4, 1]
print(selection_sort(nums)) 